### PR TITLE
CAMEL-20056: make it possible to set multiple headers at once.

### PR DESCRIPTION
**Pull request by klease**
Supported in Java, Yaml and Xml DSL.

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by klease** [comment]
Failure in maven test due to org.apache.camel.support.jsse.TrustManagersParametersTest.testPropertyPlaceholders. I think this is due to parallel tests which are writing the same file in the target directory.

**Comment by davsclaus** [comment]
> Failure in maven test due to org.apache.camel.support.jsse.TrustManagersParametersTest.testPropertyPlaceholders. I think this is due to parallel tests which are writing the same file in the target directory.

Yes it seems to be - we need to either isolate these tests or fix them so they have their private files

File: core/camel-core-model/src/main/java/org/apache/camel/model/SetHeadersDefinition.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import jakarta.xml.bind.annotation.XmlAccessorType;
+import jakarta.xml.bind.annotation.XmlElementRef;
+import jakarta.xml.bind.annotation.XmlRootElement;
+
+import org.apache.camel.Expression;
+import org.apache.camel.builder.ExpressionBuilder;
+import org.apache.camel.spi.Metadata;
+
+/**
+ * A sequence of setHeader operations.
```
Comment (davsclaus) [inline core/camel-core-model/src/main/java/org/apache/camel/model/SetHeadersDefinition.java:34]: Can we update this to better tell the user what it does, something like:

Allows to set multiple headers at once on the message.

**Review by davsclaus — COMMENTED** [review]
We need documentation in a file setHeaders-eip.adoc and in the camel-core-engine - copy the existing setHeader and modify this file.

**Comment by davsclaus** [comment]
Much better description thanks - just noticed a little typo in "st the same time"

File: core/camel-core-model/src/main/java/org/apache/camel/model/SetHeadersDefinition.java (hunk 1 above)
Comment (klease) [reply core/camel-core-model/src/main/java/org/apache/camel/model/SetHeadersDefinition.java]: setHeaders-eip.adoc added and description updated.

**Comment by klease** [comment]
> Much better description thanks - just noticed a little typo in "st the same time"

Rats, silly me! That will change all the generated files again :-(.  By the way, I can't figure out how the nav.doc for the eips is generated since building docs does the others but not that one.

**Comment by davsclaus** [comment]
> > Much better description thanks - just noticed a little typo in "st the same time"
> 
> Rats, silly me! That will change all the generated files again :-(. By the way, I can't figure out how the nav.doc for the eips is generated since building docs does the others but not that one.

You can try (if this works)

cd docs
mvn clean install

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
Also it would be good to add a TIP in setHeader doc to link to this new one so users can more easily be aware of this variation

**Comment by klease** [comment]
Thanks for committing Claus. It looks like the generation of the nav.adoc for the EIPs is broken. At least it's not done by running clean install in docs and it appears that that this is due to a not very recent change in gulpfile.js

**Comment by davsclaus** [comment]
Okay maybe @zregvart can help out

**Comment by zregvart** [comment]
@klease can you file a JIRA for the issue you're facing? I can run `./mvnw -pl :docs` successfully on my end.

**Comment by klease** [comment]
@zregvart CAMEL-20135

**Pull request merged**